controllers/storagecluster/external_resources.go

```diff
@@ -354,20 +354,21 @@ func (r *StorageClusterReconciler) createExternalStorageClusterResources(instanc
 			} else if d.Name == cephRgwStorageClassName {
 				rgwEndpoint := d.Data[externalCephRgwEndpointKey]
 				if err := checkEndpointReachable(rgwEndpoint, 5*time.Second); err != nil {
-					r.Log.Error(err, "RGW endpoint is not reachable.", "RGWEndpoint", rgwEndpoint)
-					return err
+					r.Log.Error(err, "RGW endpoint is not reachable. Will skip creating objectstore and RGW Storage Class", "RGWEndpoint", rgwEndpoint)
```

Comment:
Is the RGW required to be online to create the rgw storage class? If the endpoint is just down temporarily, this means the storage class won't be created as expected. Why not still continue creating the storage class?

Reply:
Storage class needs the `cephobjectstorename` as a parameter,

So if don't create objectstore at the start because of rgw-endpoint not reachable, then it won't be created later as this code I think execute once at the very start for initializing the external cluster.

And also as we are using the objectstorename to create the SC so I think that would be also needed later for the obc creation internallly

Reply:
And as we don't send the error to the reconcile I am not sure it will reconcile

Reply:
Before we proceed with this change, let's finalize the discussion on the BZ